sdk/kiosk/src/bcs.ts

```diff
@@ -1,66 +1,37 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { ObjectOwner, bcs } from '@mysten/sui.js';
-
-/**
- * The Kiosk object fields (for BCS queries).
- */
-export type Kiosk = {
-  id: string;
-  profits: string;
-  owner: string;
-  itemCount: number;
-  allowExtensions: boolean;
-};
+import { bcs } from '@mysten/sui.js';
+import {
+  KIOSK_PURCHASE_CAP,
+  KIOSK_TYPE,
+  TRANSFER_POLICY_CREATED_EVENT,
```

Comment:
It would be great if we had a function to create generic events of kind: transferPolicyCreated(type): `0x2:....::Created<${type}>`. Wonder what you think?

Reply:
@damirka Could you elaborate a bit on that?